tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -507,21 +499,30 @@ private void HandleUdsStats()
             }
         }
 
-        private void HandleUdsTraces()
+        private async Task HandleUdsTraces()
         {
             while (!_cancellationTokenSource.IsCancellationRequested)
             {
                 try
                 {
-                    using (var handler = _udsTracesSocket.Accept())
+                    using var handler = await _udsTracesSocket.AcceptAsync();
+                    using var stream = new NetworkStream(handler);
+
+                    var request = await MockHttpParser.ReadRequest(stream);
+                    HandlePotentialTraces(request);
+                    await stream.WriteAsync(GetResponseBytes());
+
+                    // Wait for client to close the connection
+                    // If you're reading this and you know about sockets:
+                    // I have NO IDEA if that's the right way to wait until the response was properly sent
+                    // If you know a better way, by all means, please replace this code.
+                    var buffer = new byte[256];
```

Comment:
Does the size of this buffer matter? I assume the `handler.Receive()` call ensures it doesn't overflow etc?

Reply:
Yeah, the size doesn't matter. And we shouldn't be able to read any data here anyway.